modules/graphman-bundle.js

```diff
@@ -140,97 +182,205 @@ module.exports = {
-        const attr = {};
-        const entityName = this.entityName(entity, attr);
+    isArrayEquals: function (left, right, fieldPath, callback) {
+        if (!Array.isArray(left) || !Array.isArray(right) || left.length !== right.length) {
+            if (callback) {
+                callback({
+                    path: fieldPath,
+                    left: left,
+                    right: right
+                });
+            }
+            return false;
+        }
 
-        if (entity.name) {
-            return attr.ref !== 'name' ? entity.name + "-" + entityName : entityName;
-        } else {
-            return entityName;
+        let equals = true;
+        for (let index = 0; index < left.length; index++) {
+            const leftItem = left[index];
+            if (typeof leftItem !== 'object') {
+                if (leftItem !== right[index]) {
+                    if (callback) {
+                        callback({
+                            path: fieldPath + "[" + index + "]",
+                            left: leftItem,
+                            right: right[index]
+                        });
+                        equals = false;
+                    } else {
+                        return false;
+                    }
+                }
+            } else {
+                if (!this.isObjectEquals(leftItem, right[index], fieldPath + "[" + index + "]", callback)) {
+                    if (callback) {
+                        equals = false;
+                    } else {
+                        return false;
+                    }
+                }
+            }
         }
-    }
-}
 
-function matchSoapResolvers(left, right) {
-    if (left.resolvers && right.resolvers) {
-        if (left.resolvers.baseUri && left.resolvers.baseUri !== right.resolvers.baseUri) {
-            return false;
+        return equals;
+    },
+
+    isObjectEquals: function (left, right, fieldPath, callback) {
+        if (Array.isArray(left)) {
+            return this.isArrayEquals(left, right, fieldPath, callback);
         }
 
-        if (Array.isArray(left.resolvers.soapActions) && Array.isArray(right.resolvers.soapActions)) {
-            if (left.resolvers.soapActions.length !== right.resolvers.soapActions.length) {
-                return false;
+        let equals = true;
+        for (const key of Object.keys(left)) {
+            const data = {
+                path: fieldPath + "." + key,
+                left: left[key],
+                right: right[key]
+            };
+
+            if (!right.hasOwnProperty(key)) { // is field itself missing?
+                if (callback) {
+                    data.right = null;
+                    callback(data);
+                    equals = false;
+                } else {
+                    return false;
+                }
+            } else if (left[key] == null && right[key] != null) { // special case: are they both nulls?
```

Comment:
condition differs from comment

Reply:
even though it doesn't appear this condition (serialization removes null fields), let's handle this properly.

Thanks.

Reply:
@uppoju Addressed it.